source/features/insert-summary-button.tsx

```diff
@@ -0,0 +1,55 @@
+import React from 'dom-chef';
+import select from 'select-dom';
+import onetime from 'onetime';
+import delegate from 'delegate';
+import features from '../libs/features';
+import * as icons from '../libs/icons';
+
+const addEvents = onetime(() => {
+	delegate('.rgh-summary-btn', 'click', addSummaryDetails)
+});
+
+function init() {
+	addEvents();
+	for (const anchor of select.all('md-task-list')) {
+		anchor.after(
+			<button type="button" class="toolbar-item tooltipped tooltipped-n rgh-summary-btn" aria-label="Add summary">
+				{icons.info()}
+			</button>
+		);
+	}
+}
+
+function addSummaryDetails(event) {
+	// Don't indent <summary> because indentation will not be automatic on multi-line content
+	const newContent = `
+		<details>
+		<summary>Details</summary>
+		${getSelection().toString()}
+		</details>
+	`.replace(/(\n|\b)\t+/g, '$1').trim();
+
+	// Inject new tags; it'll be undoable
+	document.execCommand('insertText', false, '\n\n' + newContent + '\n\n');
```

Comment:
The `\n\n` is necessary because that's how most "block" items behave in GH's toolbar, however sometimes they're not necessary. Up to you (whoever _you_ are) to implement a "sometimes" detector (or drop `\n\n`)

Reply:
Oh cool, I wasn't aware of that! Thanks for the context!